File: apps/mobile/widgets/ScorevaLive.swift
import SwiftUI
import WidgetKit

/// Scoreva Live Activity + home-screen widget.
/// Wired in the EAS production native project (`eas build`). Expo Go renders the
/// same compact tracker via `screens/WidgetPreviewScreen`.

struct ScorevaEntry: TimelineEntry {
  let date: Date
  let home: String
  let away: String
  let homeScore: Int
  let awayScore: Int
  let minute: Int
}

struct ScorevaProvider: TimelineProvider {
  func placeholder(in context: Context) -> ScorevaEntry {
    ScorevaEntry(date: Date(), home: "ARS", away: "CHE", homeScore: 2, awayScore: 1, minute: 67)
  }
  func getSnapshot(in context: Context, completion: @escaping (ScorevaEntry) -> Void) {
    completion(placeholder(in: context))
  }
  func getTimeline(in context: Context, completion: @escaping (Timeline<ScorevaEntry>) -> Void) {
    completion(Timeline(entries: [placeholder(in: context)], policy: .after(Date().addingTimeInterval(30))))
  }
}

struct ScorevaWidgetEntryView: View {
  var entry: ScorevaEntry
  var body: some View {
    VStack(alignment: .leading, spacing: 6) {
      Text("LIVE \(entry.minute)’").font(.caption.monospaced()).foregroundStyle(Color(red: 0.84, green: 1, blue: 0.24))
      HStack {
        Text(entry.home)
        Spacer()
        Text("\(entry.homeScore)").font(.title.monospaced())
      }
      HStack {
        Text(entry.away)
        Spacer()
        Text("\(entry.awayScore)").font(.title.monospaced())
      }
    }
    .padding()
    .containerBackground(Color(red: 0.03, green: 0.035, blue: 0.05), for: .widget)
  }
}

@main
struct ScorevaWidgets: Widget {
  var body: some WidgetConfiguration {
    StaticConfiguration(kind: "ScorevaLive", provider: ScorevaProvider()) { entry in
      ScorevaWidgetEntryView(entry: entry)
    }
    .configurationDisplayName("Scoreva Live")
    .description("Live match on the home screen.")
    .supportedFamilies([.systemSmall, .systemMedium])
  }
}

File: apps/mobile/widgets/ScorevaLive.kt
package app.scoreva.widget

import androidx.glance.appwidget.GlanceAppWidget
import androidx.glance.appwidget.GlanceAppWidgetReceiver
import androidx.glance.appwidget.provideContent
import androidx.glance.GlanceId
import androidx.glance.GlanceModifier
import androidx.glance.layout.Column
import androidx.glance.text.Text
import android.content.Context

/** Android home-screen widget. Included in the EAS production native project. */
class ScorevaLiveWidget : GlanceAppWidget() {
    override suspend fun provideGlance(context: Context, id: GlanceId) {
        provideContent {
            Column(modifier = GlanceModifier) {
                Text("LIVE 67'")
                Text("ARS  2")
                Text("CHE  1")
            }
        }
    }
}

class ScorevaLiveReceiver : GlanceAppWidgetReceiver() {
    override val glanceAppWidget: GlanceAppWidget = ScorevaLiveWidget()
}
